src/components/AppComponent.js

```diff
@@ -0,0 +1,176 @@
+import CustomComponent from "../abstracts/CustomComponent";
+import HeaderComponent from "./AppHeaderComponent";
+import MovieListComponent from "./movie/MovieListComponent";
+import MoreButtonComponent from "./element/MoreButtonComponent";
+import TitleComponent from "./element/TitleComponent";
+import transformMovieItemsType from "../util/MovieList";
+import { API_KEY } from "../constants/key";
+import {
+  ACTION,
+  REQUEST_URL,
+  SEARCH_WARNING,
+  TITLE,
+} from "../constants/constants";
+
+export default class AppComponent extends CustomComponent {
+  #nextPage = 1;
+  #totalPage;
+  #$movieList;
+  #$movieListTitle;
+  #$searchInput;
+
+  render() {
+    super.render();
+
+    this.#$movieList = this.querySelector("movie-list");
+    this.#$movieListTitle = this.querySelector("movie-list-title");
+    this.#$searchInput = this.querySelector("input");
+
+    this.popularListInit();
+    this.renderListByData(ACTION.POPULAR);
+    this.changeMoreButtonAction(ACTION.MORE_POPULAR);
+  }
+
+  async fetchAPI(actionType) {
+    switch (actionType) {
+      case ACTION.POPULAR:
+        return await fetch(
+          `${REQUEST_URL}/movie/popular?api_key=${API_KEY}&language=ko-KR&page=${
+            this.#nextPage
+          }`,
+          { method: "GET" }
+        );
+      case ACTION.SEARCH:
+        return await fetch(
+          `${REQUEST_URL}/search/movie?api_key=${API_KEY}&language=ko-KR&query=${
+            this.#$searchInput.value
+          }&page=${this.#nextPage}&include_adult=false`,
+          { method: "GET" }
+        );
+      default:
+        return null;
+        break;
+    }
```

Comment:
이건 swtich-case 관련 소소한 팁인데 아래처럼도 작성 가능하답니다

```suggestion
      case ACTION.SEARCH:
      default:
        return await fetch(
          `${REQUEST_URL}/search/movie?api_key=${API_KEY}&language=ko-KR&query=${
            this.#$searchInput.value
          }&page=${this.#nextPage}&include_adult=false`,
          { method: "GET" }
        );
    }
```

Reply:
아 감사합니다!! 활용해보겠습니다 😁